WebContent/zip-fs.js

```diff
@@ -301,7 +298,9 @@
 			that.id = fs.entries.length;
 			that.parent = parent;
 			that.children = [];
-			that.zipVersion = params.zipVersion || 0x14;
+			that.options = params.options || {};
+			that.options.version = params.zipVersion || 0x14;
+			that.options.directory = that.directory;
```

Comment:
And change these lines to

```js
that.compressionLevel = params.compressionLevel;
```

(your current method has an issue: it takes and modifies the user-supplied `options` object)